### PR TITLE
✨ feat(k8sgpt-rbac.yaml): add new RBAC configuration for k8sgpt operator

### LINKED ISSUE #259: [Feature]: Migrate all RBAC related resources to Helm chart's templates
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

None

### Problem Description

Both manager's and k8sgpt's roles have to match in order for the former to grant permissions to K8sgpt cluster role or add the verb `escalate`. Following the Least privileged access idiom, would be great if we move away from the existing way of managing roles

### Solution Description

Lift and shift the cluster role, bindings and service account from operator's code to Helm chart's tempates

### Benefits

To be more transparent to the end users and also avoid situations like K8s API [restrictions](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update) with roles which caused issues in the latest `v0.0.22`

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by MateSousa**
Closes #259

## 📑 Description

Migrate all RBAC related resources to Helm chart's templates

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: chart/operator/templates/k8sgpt-rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "chart.fullname" . }}-operator-role
```
Comment (arbreezy) [inline chart/operator/templates/k8sgpt-rbac.yaml:4]: I think we need a better naming to designate that this is for K8sGPT workload and not operator related

File: chart/operator/templates/k8sgpt-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - apiextensions.k8s.io
+    resources:
+      - '*'
+    verbs:
+      - '*'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ include "chart.fullname" . }}-operator-rolebinding
```
Comment (arbreezy) [inline chart/operator/templates/k8sgpt-rbac.yaml:28]: same here regarding naming, suggesting to name it `{{ include "chart.fullname" . }}-k8sgpt`

File: chart/operator/templates/k8sgpt-rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    {{- include "chart.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ include "chart.fullname" . }}-operator-role
+subjects:
+  - kind: ServiceAccount
+    name: "k8sgpt"
+    namespace: {{ .Release.Namespace }}
+---
```
Comment (arbreezy) [inline chart/operator/templates/k8sgpt-rbac.yaml:39]: A generic comment, you should split these K8s resources to different template yaml files

**Review by arbreezy — CHANGES_REQUESTED** [review]

**Comment by arbreezy** [comment]
@MateSousa you want also to remove the permissions of the operator's cluster [role](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/chart/operator/templates/manager-rbac.yaml#L58) cause operator won't manage anymore the k8sgpt's role creation right?

**Comment by MateSousa** [comment]
> @MateSousa you want also to remove the permissions of the operator's cluster [role](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/chart/operator/templates/manager-rbac.yaml?rgh-link-date=2023-11-26T21%3A36%3A12Z#L58) cause operator won't manage anymore the k8sgpt's role creation right?

@arbreezy you're right, i just removed

**Comment by arbreezy** [comment]
Apologies @MateSousa , I forgot that we didn't merge this PR; overall looks good !

can you please resolve the conflicts and I will test a bit more before we merge it

**Comment by MateSousa** [comment]
@arbreezy done

**Review by arbreezy — CHANGES_REQUESTED** [review]
@MateSousa , I think you override an existing functionality here.

File: pkg/resources/k8sgpt.go [hunk 4]
```diff
@@ -433,31 +330,7 @@ func Sync(ctx context.Context, c client.Client,
...
-
-	svc, er := GetService(config)
-	if er != nil {
-		return er
-	}
-
-	objs = append(objs, svc)
-
-	deployment, er := GetDeployment(config, outOfClusterMode, c)
+	deployment, er := GetDeployment(config)
```
Comment (arbreezy) [inline pkg/resources/k8sgpt.go:333]: ^

**Comment by arbreezy** [comment]
Hey @MateSousa I am happy to properly resolve the previous conflicts and get this merged.

**Comment by MateSousa** [comment]
Hey @arbreezy, sorry for delay, I was sick... I'm going to fix that today.

**Comment by arbreezy** [comment]
> Hey @arbreezy, sorry for delay, I was sick... I'm going to fix that today.

@MateSousa hope you are feeling better !

**Comment by arbreezy** [comment]
@MateSousa, happy to help with this PR if you don't have time, it's almost ready to be merged

**Comment by MateSousa** [comment]
@arbreezy hey, it looks like I made a huge mess, I'm going to create a new PR, sorry :(

**Pull request closed without merging**